src/luarocks/fs/win32/tools.lua

```diff
@@ -159,6 +159,33 @@ function tools.is_file(file)
    return fs.execute(fs.Q(vars.TEST).." -f", file)
 end
 
+--- Set permissions for file or directory
+-- @param filename string: filename whose permissions are to be modified
+-- @param mode string ("read" or "exec"): permission to set
+-- @param scope string ("user" or "all"): the user(s) to whom the permission applies
+-- @return boolean or (boolean, string): true on success, false on failure,
+-- plus an error message
+function tools.set_permissions(filename, mode, scope)
+   assert(filename and mode and scope)
+
+   local who, what
+   if scope == "user" then
+      who = "%USERNAME%"
+   elseif scope == "all" then
```

Comment:
I'm not sure if it's the case or not, but please double-check that when the scope is set to "user" then everyone else is being _denied_ access (I know this is correct in your code for Unix with the `00` in the values, I'm not sure on Windows)

This is used in the code to set permissions of the API key for uploads, to make sure that only its owner can see it.